lib/widgets/message_list.dart

```diff
@@ -689,28 +688,9 @@ class _MarkAsReadWidgetState extends State<MarkAsReadWidget> {
 
   void _handlePress(BuildContext context) async {
     if (!context.mounted) return;
-
-    final store = PerAccountStoreWidget.of(context);
-    final connection = store.connection;
-    final useLegacy = connection.zulipFeatureLevel! < 155;
     setState(() => _loading = true);
-
-    try {
-      await markNarrowAsRead(context, widget.narrow, useLegacy);
-    } catch (e) {
-      if (!context.mounted) return;
-      final zulipLocalizations = ZulipLocalizations.of(context);
-      showErrorDialog(context: context,
-        title: zulipLocalizations.errorMarkAsReadFailedTitle,
-        message: e.toString()); // TODO(#741): extract user-facing message better
-      return;
-    } finally {
-      setState(() => _loading = false);
-    }
-    if (!context.mounted) return;
-    if (widget.narrow is CombinedFeedNarrow && !useLegacy) {
-      PerAccountStoreWidget.of(context).unreads.handleAllMessagesReadSuccess();
```

Comment:
In the first commit:
f2dec7f84 actions [nfc]: Port generic error handling to `markNarrowAsRead`

there looks to be a change in when these lines get run. Previously we'd run them whenever there wasn't an exception; now we do so only when `markNarrowAsRead` returns true. And there are a couple of cases where it didn't throw an exception, but now returns false. That makes the commit not NFC.

I think those behavior changes are probably good, but let's separate them from the NFC change which affects more lines of code. A first commit can make the NFC change of moving that try/catch inside the function — to make it NFC, it should return false just when the old version would have thrown. Then a second commit can start returning false in more cases.

(Then continue with the next commit:
d4d26155b actions [nfc]: Port `handleAllMessagesReadSuccess` to `markNarrowAsRead`

That commit being NFC relies on having already made this non-NFC change.)